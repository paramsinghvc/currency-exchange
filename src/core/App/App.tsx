import React from "react";
import styled from "@emotion/styled";
import { BrowserRouter as Router, Route, Redirect, Switch } from "react-router-dom";

import Home from "scenes/Home";
import Transactions from "scenes/Transactions";
// import Shell from "./components/Shell";
import createStore from "core/utils/createStore";
import { createStoreContext } from "@mollycule/redux-hook";
import { IRootState } from "shared/types";

const store = createStore();
const { Provider } = createStoreContext<IRootState>();

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Route path="/" exact component={Transactions} />
          <Route path="/exchange" component={Home} />
        </Switch>
      </Router>
    </Provider>
  );
};

export default App;
