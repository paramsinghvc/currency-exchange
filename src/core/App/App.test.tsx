import React from "react";
import ReactDOM, { render } from "react-dom";
import App from "./App";
import renderer, { act as rendererAct } from "react-test-renderer";
import { act } from "react-dom/test-utils";

it("matches the snapshot", () => {
  const tree = renderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  act(() => {
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
