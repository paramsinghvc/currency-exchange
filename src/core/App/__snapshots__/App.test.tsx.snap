// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<main
  className="css-o8uaij"
>
  <header
    className="css-uf1ume"
  >
    <section
      className="css-m6ba0"
      role="button"
    >
      <span />
      <span />
      <span />
    </section>
  </header>
  <main
    className="css-13qacuc"
  >
    <section
      className="css-k68f27"
    />
    <section
      className="css-194lopa"
    >
      <section
        className="css-1oskuwq"
      >
        <div
          className="css-1qvptwv"
          onChange={[Function]}
          type="text"
          value="15.34"
        >
          <input
            className="css-5qep15"
            onChange={[Function]}
            type="text"
            value="15.34"
          />
        </div>
        <div
          className="css-1x7hx5v"
          onClick={[Function]}
        >
          <p
            className="css-1b5o5hi"
          >
            
          </p>
          <p
            className="css-1j7n4jp"
          >
            
          </p>
        </div>
      </section>
      <div
        className="css-aijkp"
      >
        <img
          className="css-bobwps"
          src="Coin.svg"
        />
      </div>
      <section
        className="css-1oskuwq"
      >
        <div
          className="css-1qvptwv"
          onChange={[Function]}
          type="text"
          value="19.73"
        >
          <input
            className="css-5qep15"
            onChange={[Function]}
            type="text"
            value="19.73"
          />
        </div>
        <div
          className="css-1x7hx5v"
          onClick={[Function]}
        >
          <p
            className="css-1b5o5hi"
          >
            
          </p>
          <p
            className="css-1j7n4jp"
          >
            
          </p>
        </div>
      </section>
    </section>
    <section
      className="css-1hppjzv"
    >
      <button
        className="css-fi2zjy"
        color="#3A539B"
        label="EXCHANGE"
        onClick={[Function]}
        size="x-large"
      >
        EXCHANGE
      </button>
    </section>
  </main>
</main>
`;
